### PR TITLE
feat: Search Decision Requirements - Controller

### LINKED ISSUE #20704: [C8 API] Implement `Decision requirement` endpoints
### Description

Decision requirement endpoints provides searching for decision requirement.

* The endpoints should be implemented as described in [REST API Developer Guide](https://github.com/camunda/camunda/blob/main/docs/rest-controller.md)
* `{place-holder}` is the placeholder for parameters like `decisionDefinitionKey`
* Decision requirement objects are defined in
  * [Decision requirement](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.pe9vw4ocuyy8)

### Endpoints

- [x] Search Decision requirements - `POST /v2/decision-requirements/search
` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.3gvs1btxwocy)
- [x] Get decision requirements by key - `GET /v2/decision-requirements/{decisionRequirementsKey}` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.xp6iv6345btw)
- [x] Get decision requirements XML - `GET /v2/decision-requirements/{decisionRequirementsKey}/xml` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.u59mvpqo1amd)

### Implementation
- [x] Service implementation - https://github.com/camunda/camunda/pull/20722
- [x] Controller implementation  - https://github.com/camunda/camunda/pull/20769
- [x] Client implementation -  https://github.com/camunda/camunda/pull/20786
- [x] Integration Tests - https://github.com/camunda/camunda/pull/20821

- [x] Get Decision Requirements by Key - https://github.com/camunda/camunda/pull/22651

### Acceptance Criteria

- [x] Unit tests
- [x] Integration tests
### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered and added to the product-hub issue risk assessment
- [ ] Cross-team dependencies have been considered

Optional:
- [ ] Design input has been collected by the assignee
### :point_right: Handover Dev to QA 

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Versions to validate:
> Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released)

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Implement `/definition-requirements/search` endpoint

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -587,6 +587,29 @@ paths:
             application/problem+json:
               schema:
                 $ref: "#/components/schemas/ProblemDetail"
+  /decision-requirements/search:
+    post:
+      tags:
+        - Decision Requirements
+      summary: Decision Requirements Search Query
```
Comment (houssain-barouni) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:594]: @tmetzke reworked the endpoints summary here https://github.com/camunda/camunda/pull/20807/files

This should be changed to `Query decision requirements (experimental)`

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -587,6 +587,29 @@ paths:
...
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DecisionRequirementsSearchQueryRequest'
```
Comment (houssain-barouni) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:600]:
```suggestion
              $ref: "#/components/schemas/DecisionRequirementsSearchQueryRequest"
```

**Review by houssain-barouni — CHANGES_REQUESTED** [review]
Great work @nathansandi !
Just two comments to address in the rest-api spec

**Comment by nathansandi** [comment]
> Great work @nathansandi ! Just two comments to address in the rest-api spec

Thank you @houssain-barouni :) The improvements/fixes on rest-api.yml was applied here https://github.com/camunda/camunda/pull/20769/commits/8e06632f188b4afb3ee80166103ef4f8094053eb

**Review by houssain-barouni — APPROVED** [review]
LGTM!

**Pull request merged**